### PR TITLE
Feature/Popular Restaurants 

### PR DESCRIPTION
**Pull request by OrenMoveo**
# Popular Restaurant Feature

## This PR includes:
- Implemented mobile and desktop responsiveness for the popular restaurant section.
- Added global constants to use a single source of truth for app width, avoiding unnecessary paddings.

## Notes
- Made design decisions for handling restaurant card overflow in desktop view.
- Resolved overflow issue by adding an inner scroll for desktop view.
- Ensured a consistent user experience across different screen sizes.

## Examples

https://github.com/OrenMoveo/Epicure/assets/156906187/224f835e-7d1e-40e0-8f72-a4364b6e8126

https://github.com/OrenMoveo/Epicure/assets/156906187/f7e41885-685e-48ab-b9f0-367ee166ff0a

### CODE REVIEW

File: src/components/PopularRestaurants/PopularRestaurant.module.scss
Comment (nitzanpap) [inline src/components/PopularRestaurants/PopularRestaurant.module.scss:1]: This SCSS module looks f*n great :fire: 
Great job, don't you think?

File: src/components/PopularRestaurants/PopularRestaurant.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                .slice(0, 3)
+                .map((restaurant, keyId) => (
+                  <RestaurantCard key={keyId} restaurant={restaurant} />
+                ))}
+          </div>
+        )}
+
+        <button
+          className={styles.allRestaurantsGotoContainer}
+          name="all-restaurant-goto-button"
```
Comment (nitzanpap) [inline src/components/PopularRestaurants/PopularRestaurant.tsx:53]: Note that `styles.allRestaurantsGotoContainer` is not quite clear enough.
Shouldn't it be something along the lines of `styles.gotoAllRestaurantsBtn`?
After all, it is indeed a *Button* and not just a container.

File: src/components/PopularRestaurants/PopularRestaurant.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        >
+          <p className={styles.allRestaurantsGotoText}>All Restaurants</p>
```
Comment (nitzanpap) [inline src/components/PopularRestaurants/PopularRestaurant.tsx:55]: Same here.

**Review by nitzanpap — APPROVED** [review]

File: src/components/PopularRestaurants/PopularRestaurant.tsx (hunk 1 above)
Comment (OrenMoveo) [reply src/components/PopularRestaurants/PopularRestaurant.tsx:53]: @nitzanpap done

**Pull request merged**